zio-schema-zio-test/shared/src/main/scala/zio/schema/DeriveGen.scala

```diff
@@ -0,0 +1,528 @@
+      f8  <- gen(caseClass21.field8.schema)
+      f9  <- gen(caseClass21.field9.schema)
+      f10 <- gen(caseClass21.field10.schema)
+      f11 <- gen(caseClass21.field11.schema)
+      f12 <- gen(caseClass21.field12.schema)
+      f13 <- gen(caseClass21.field13.schema)
+      f14 <- gen(caseClass21.field14.schema)
+      f15 <- gen(caseClass21.field15.schema)
+      f16 <- gen(caseClass21.field16.schema)
+      f17 <- gen(caseClass21.field17.schema)
+      f18 <- gen(caseClass21.field18.schema)
+      f19 <- gen(caseClass21.field19.schema)
+      f20 <- gen(caseClass21.field20.schema)
+      f21 <- gen(caseClass21.field21.schema)
+    } yield caseClass21.construct(f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12, f13, f14, f15, f16, f17, f18, f19, f20, f21)
+
+  private def genCaseClass22[A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19, A20, A21, A22, Z](
+    caseClass22: Schema.CaseClass22[A1, A2, A3, A4, A5, A6, A7, A8, A9, A10, A11, A12, A13, A14, A15, A16, A17, A18, A19, A20, A21, A22, Z]
+  ): Gen[Random with Sized, Z] =
+    for {
+      f1  <- gen(caseClass22.field1.schema)
+      f2  <- gen(caseClass22.field2.schema)
+      f3  <- gen(caseClass22.field3.schema)
+      f4  <- gen(caseClass22.field4.schema)
+      f5  <- gen(caseClass22.field5.schema)
+      f6  <- gen(caseClass22.field6.schema)
+      f7  <- gen(caseClass22.field7.schema)
+      f8  <- gen(caseClass22.field8.schema)
+      f9  <- gen(caseClass22.field9.schema)
+      f10 <- gen(caseClass22.field10.schema)
+      f11 <- gen(caseClass22.field11.schema)
+      f12 <- gen(caseClass22.field12.schema)
+      f13 <- gen(caseClass22.field13.schema)
+      f14 <- gen(caseClass22.field14.schema)
+      f15 <- gen(caseClass22.field15.schema)
+      f16 <- gen(caseClass22.field16.schema)
+      f17 <- gen(caseClass22.field17.schema)
+      f18 <- gen(caseClass22.field18.schema)
+      f19 <- gen(caseClass22.field19.schema)
+      f20 <- gen(caseClass22.field20.schema)
+      f21 <- gen(caseClass22.field21.schema)
+      f22 <- gen(caseClass22.field22.schema)
+    } yield caseClass22.construct(f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12, f13, f14, f15, f16, f17, f18, f19, f20, f21, f22)
+  // scalafmt: { maxColumn = 120 }
+
+  private def genGenericRecord(record: Schema.GenericRecord): Gen[Random with Sized, ListMap[String, _]] =
+    record.structure
+      .foldLeft[Gen[Random with Sized, ListMap[String, _]]](Gen.const(ListMap.empty)) {
+        case (genListMap, field) =>
+          for {
+            listMap <- genListMap
+            value   <- gen(field.schema)
+          } yield listMap.updated(field.label, value)
+      }
+
+  private def genSequence[Z, A](seq: Schema.Sequence[Z, A]): Gen[Random with Sized, Z] =
+    Gen.chunkOf1(gen(seq.schemaA)).map(nec => seq.fromChunk(nec.toChunk))
+
+  private def genMap[K, V](map: Schema.MapSchema[K, V]): Gen[Random with Sized, Map[K, V]] =
+    Gen.mapOf1(gen(map.ks), gen(map.vs))
```

Comment:
For these two I would not restrict to single-element chunks (at the very least we should sometimes generate an empty Chunk). So maybe a chunk bounded by 0 and 2? I know in other places we run into severe performance issues if  we don't restrict to small(ish) collection sized.

Reply:
Yes, I was concerned about chunk's arbitrary size and that's why I did single-element chunks).
No problem, I will bound the chunk by 0 and 2